### ISSUE TITLE
Bug: InfoMax3DModular pretraining fails to update model3d parameters (3D network remains untrained)

### ISSUE THREAD

**Issue by anushkagupta200615-jpg** [deepchem/deepchem#5048]
In `deepchem/models/torch_models/gnn3d.py`, the `InfoMax3DModular` model implements the 3D Infomax pretraining algorithm. This algorithm relies on contrastive learning by maximizing the mutual information between a 2D GNN (`model2d`) and a 3D GNN (`model3d`).

However, there is a critical architectural flaw that prevents the 3D network from ever being trained. 

Because `InfoMax3DModular` inherits from `ModularTorchModel` (and subsequently `TorchModel`), the PyTorch optimizer is implicitly constructed using `self.model.parameters()`. But in `gnn3d.py`, `build_model()` explicitly returns *only* `model2d`:

```python
    def build_model(self):
        # FIXME For pretraining task, both model2d and model3d but the super class
        # can't handle two models for contrastive learning, hence we pass only model2d
        return self.components['model2d']
```

Because `model3d` is not included in the returned model, **its parameters are never registered with the PyTorch optimizer**. During `fit()`, the loss is computed correctly using `model3d` in `loss_func`, but when `.backward()` is called and the optimizer steps, **only `model2d` is updated**. 

As a result, `model3d` remains completely frozen at its random initialization. The 2D network is effectively learning to match random noise, which entirely defeats the purpose of the 3D Infomax pretraining algorithm.

## To Reproduce

1. Initialize `InfoMax3DModular` with `task='pretraining'`.
2. Capture a snapshot of the weights of `model3d` before training.
3. Call `model.fit(dataset)` on a sample dataset.
4. Check the weights of `model3d` after training — they will be identical to the pre-training weights, proving that no gradients were applied to the 3D network during the contrastive learning phase.

## Expected behavior

During the `pretraining` task, the optimizer should receive the parameters for both `model2d` and `model3d` so that the contrastive learning updates both networks simultaneously. 

## Environment

* OS: Any
* Python version: Any
* DeepChem version: Master branch (latest)
* RDKit version (optional): N/A
* TensorFlow version (optional): N/A
* PyTorch version (optional): Any
* Any other relevant information: This is a static architecture bug in `gnn3d.py`.

## Additional context

There are a couple of ways to fix this without breaking the `TorchModel` API:
1. **Wrapper Module**: Modify `build_model()` to return a thin `nn.Module` wrapper during pretraining. This wrapper would register both `model2d` and `model3d` as sub-modules (so `self.model.parameters()` yields the parameters of both), but its `forward()` method would simply route to `model2d`.
2. **Custom Optimizer Definition**: Override the optimizer initialization logic in `ModularTorchModel` or `InfoMax3DModular` to explicitly chain and pass both parameters: `itertools.chain(self.components['model2d'].parameters(), self.components['model3d'].parameters())`.
```

**Comment by anushkagupta200615-jpg** [comment]
Hi maintainers, I've just opened a Pull Request to fix this issue! #5050 

I created a `_PretrainingWrapper` module inside `gnn3d.py` that encapsulates both `model2d` and `model3d`. Now, when `InfoMax3DModular` runs in `pretraining` mode, `build_model()` returns this wrapper. This ensures that PyTorch's native optimizer registration correctly captures the `.parameters()` for both networks. 

I've also added a new unit test in `test_gnn3d.py` that explicitly asserts that both `model2d` and `model3d` parameters are correctly registered with the model's optimizer.

I would love to get your thoughts and a review on the changes whenever you have a moment!

**Comment by Ritesh-1127** [comment]
I'm going to look why the optimizer isn't registering the 3D parameters. I'll test a fix and aim to have a PR up shortly.
